crates/bevy_winit/src/web_resize.rs

```diff
@@ -20,13 +20,14 @@ struct ResizeEvent {
     window_id: WindowId,
 }
 
+#[derive(Resource)]
 pub(crate) struct CanvasParentResizeEventChannel {
     sender: Sender<ResizeEvent>,
     receiver: Receiver<ResizeEvent>,
 }
 
 fn canvas_parent_resize_event_handler(
-    winit_windows: Res<WinitWindows>,
+    winit_windows: NonSend<WinitWindows>,
```

Comment:
This is an odd one.

This module only exists with the `wasm` feature enabled,
but `WinitWindows` is purposefully made non-send since some of it's API's need to be run on the main thread.
And, `Res` already required `Send + Sync`, So I realized after your comment this shouldn't have compiled to begin with.

I looked around a bit and..
Turns out if the `wasm` feature is enabled 2 unsafe impls for Send and Sync are implemented on `WinitWindows` relying on the fact that wasm is single-threaded :upside_down_face:

I have no idea why this didn't use `NonSend` to begin with.
I removed the unsafe impls and it compiles fine. Guess I'll have to test it?

Reply:
I've no idea. My hunch is that it should be fine, but it's a heavy case of [Chesterton's fence](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence) here, I'd like to hear the reasoning behind the `unsafe impl` first. Maybe `git blame` on the unsafe impl could help?

Reply:
Just tested and example in wasm and it ran fine.
*(First time setting up wasm for Bevy. Super easy!)*

I did find the PR via git blame #503 and
It's 2 years old and didn't state exactly why it was needed and I can't find a reason for it in the changes.

As far as I know `Send` and `Sync` only exist at compile time, zero-cost abstractions and all that, so if it compiles it should be fine :shrug:
I'll remove it for now but I'll leave this conversation open.

Reply:
I don't love the unsafe impl, but we should pull that change out into it's own PR to make sure it gets proper review.